RULES/cc/test.go

```diff
@@ -0,0 +1,28 @@
+package cc
+
+import (
+	"dbt-rules/RULES/core"
+	"fmt"
+)
+
+type Test struct {
+	Lib Library
+}
+
+func (test *Test) Build(ctx core.Context) core.BuildOutput {
+	out := ctx.Cwd().WithFilename("test")
+
+	ctx.SetBuildOption(ToolchainParam, TestToolchainA)
+	x1 := ctx.Build(&test.Lib).Output()
+	ctx.SetBuildOption(ToolchainParam, TestToolchainB)
+	x2 := ctx.Build(&test.Lib).Output()
```

Comment:
Doesn't this introduce global state through the ctx? So the person writing the rules needs to have a convention in place as to whether the ctx is caller-save or callee-save. Maybe we should make the ctx immutable, and SetBuildOption returns a copy. This way changes to the context don't propagate up the Build() chain. It would introduce some overhead, though.